.github/actions/run-a-specific-acceptance-test/action.yml

```diff
@@ -1,72 +1,81 @@
-name: Run Acceptance Test
-description: Runs the acceptance tests using python installation.
+name: Run a specific acceptance test
+description: Runs a specific acceptance test-suite that is given as an input.
 
 inputs:
   test-suite:
-    description: Acceptance test suite to run.
+    description: 'The acceptance test suite to run(e.g. "interested-donor/make-a-donation").'
     required: true
-  mobile:
+  viewport-type:
     description: Whether to run the tests on mobile or desktop.
     required: true
-    type: boolean
-  screen-recording:
+    type: choice
+    options:
+      - mobile
+      - desktop
+  enable-screen-recording:
     description: Whether to record the screen during the test.
     required: true
     type: boolean
-  modified-suite-name:
-    description: A unique suite name without spaces to use in the artifacts uploads. If not provided, the test data will not be uploaded.
+  modified-suite-name-for-uploads:
+    description: A unique suite name without spaces and forward-slashes to use for the artifacts uploads. If not provided, the test data will not be uploaded.
 
 runs:
   using: "composite"
   steps:
-    - name: Run ${{ inputs.mobile && 'Mobile' || 'Desktop' }} Acceptance Test ${{ inputs.test-suite }}
+    - name: Check if modified-suite-name-for-uploads contains invalid characters.
+      run: |
+        if [[ ! "${{ inputs.modified-suite-name-for-uploads }}" =~ ^[a-zA-Z0-9_-]+$ ]]; then
+          echo "Modified suite name for uploads contains one of the invalid characters ('/', '\', '_')"
+          exit 1
```

Comment:
Just to check, if this exits, then does the run stop here or are the "if (always() && x = failure)" steps below run as well?

Reply:
I need to check this and will add to proof of changes. You can keep it open until then.

Also, I think we should switch to `failure()` instead of `always()`. Because, I have seen that when we cancel workflow manually, then the steps with `always()` may run if the condition is satisfied.

Reply:
Hm, it's a bit tricky because if e.g. an upload action fails you still want those actions to run. So the always() has a purpose, I think...

There may not be a perfect solution here. Maybe evaluate some options and pick the least bad one?

Reply:
@seanlip I have changed it to `!cancelled()`. Also, I have added proof of changes showing that below steps run too.